### ISSUE TITLE
Correct Dependencies for ArchLinux please

### ISSUE THREAD

**Issue by schmitmd** [ProtonVPN/linux-app#67]
>  We are happy to answer your questions about the code or discuss technical ideas. 
 
Please complete the following checklist (by adding [x]):
- [x] I have searched open and closed issues for duplicates
- [x] This isn't a feature request
- [x] This is not a report about my app not working as expected

----------------------------------------

It is not a safe assumption that a user has gnome-keyring or kwallet and uses NetworkManager as their main networking solution.  A _lot_ of Arch users set things up very minimally and do not want programs that they themselves don't have an explicit need for. If such programs are necessary for a tool they _do_ have a need for, it is reasonable to expect that they will be installed as dependencies.

My experience in recent days is below.  I hope it explains how I got here and why I think this relatively simple change is important:
1. If you use Netctl rather than NetworkManager, ProtonVPN does not work and the error output does not make it immediately clear that the problem is simply "you must use NetworkManager and no other networking solution for this to function correctly."
2. If you do not have nm-applet _installed and running_, both CLI and GUI do essentially the same.  Minimal intelligible error output to explain _why_ you're not getting a connection and instead just seeing a hung attempt.
3. If you've gotten past 1) and 2), but have not installed and started a system keyring (gnome-keyring or kwallet), you will get prompted for a password by nm-applet _every time you try to connect_.  Entering your openvpn/ikev2 password works fine for this, but makes the whole thing pretty tedious.  Installing, configuring, and starting a keyring gets you fully up and running in both GUI and CLI.

The above three steps took me a while to diagnose and fix. While all of the information I truly needed _is_ specified in [this post](https://protonvpn.com/support/official-linux-client-arch/), almost all of it could have been avoided by simply adding a dependency entry for the package or putting in a clear error asking if these 3 things are in the necessary state.

Basically, can you please edit the PKGBUILD file(s) to include gnome-keyring or kwallet and nm-applet as **required** dependencies?

I realize I'm creating this issue semi-outside of the normal process flow and I apologize for that, but I really think this is a simple fix that would save a lot of people a lot of time.
Relevant bug reports that helped me get here (in addition to the link above): #28 , #12, [linux-cli#6](https://github.com/ProtonVPN/linux-cli/issues/6).

**Comment by schmitmd** [comment]
Perhaps depend on `org.freedesktop.secrets` to cover "either-or" case for keyrings? Probably need to verify that KeepassXC works correctly as well for this; it is (the only, at the moment, afaict) another pkg that provides this.

At the least, any/all of these could be added to "optional deps"

**Comment by edl2** [comment]
Thank you sooo much. nowhere in any documentation did it say this, and it never gave any useful output, it just hang. It requires gnome-keyring/kwallet and nm-applet to be installed and running.

**Comment by dron1885** [comment]
> KeepassXC works correctly as well for this

Can verify, personally use KeppasXC Secret Service integration for over a year, and PVPN works like a charm with it.

**Comment by dron1885** [comment]
> > KeepassXC works correctly as well for this
> 
> Can verify, personally use KeppasXC Secret Service integration for over a year, and PVPN works like a charm with it.

It seems that about two months ago protonvpn GUI and cli stopped logging in with KeepassXC SS, only KDE and gnome are working.

**Comment by Hielyr** [comment]
> Basically, can you please edit the PKGBUILD file(s) to include gnome-keyring or kwallet and nm-applet as **required** dependencies?

Thank you very much for putting in the work to discover this. 

I had the issue where without these I would attempt to connect the CLI or GUI would freeze for a bit, and then tell me "unknown reason occured."

I already had kwallet, but not network-manager-applet (Arch). Immediately after installing it and running nm-applet, the same thing would happen with the hang and the unhelpful message. After a reboot however, I was able to launch nm-applet and attempt a connection and it began working.

**Comment by flowHater** [comment]
@Hielyr maybe a bit late, but you can print the error, in the code. If the error is unknown, it will print a useles message. But instead you can add `print(e)` after `except (exceptions.ProtonVPNException, Exception) as e:` in `/lib/python3.10/site-packages/protonvpn_cli/cli_wrapper.py` (for me).It will print you the real internal error.
That's how I discover that protonVpn for Linux depends on NetworkManager which I haven't because it's useless (for me at least).

**Comment by cenktaskin** [comment]
Out of curiosity: what is the replacement for you NM, networkd? 
And are you able to use protonvpn somehow with it?
